### PR TITLE
Release/1.10.0

### PR DESCRIPTION
**Pull request by thoma202**
Hot fixes for ebay module. 

List of hotfixes : 

- Pagination on categories
- Double image product
- Database on 32 bits blocks store category configuration
- Send tracking number
- Use of $params['id_product'] if $params['product']->id does not exists
- Fix on CRON usages
- Fix when API logs are enabled and user changes states
- Fix on sending shipping code
- Fix when an order has two same product with two different variations 
- Alert when user tries to connect with email and not ebay username
- Deletes javascript format on description
- Send store category on revise fixed price item
- Use POST on jquery ajax call to use https or http. 
- Fixed when installing the module without configurating it.

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
Hi @thoma202,

Even with PR, we still get the following error when we try to add a product on a shop without ebay configured:
```
Fatal error: Call to a member function getConfiguration() on a non-object in /var/www/html/PrestaShop/modules/ebay/classes/EbayCategoryConfiguration.php on line 39
```

File: classes/TotFormat.php [hunk 1]
```diff
@@ -124,7 +124,7 @@ public static function formatDescription($desc)
             $desc = preg_replace('/<script\b[^>]*>(.*?)<\/script>/is', "", $desc);      
             $desc = preg_replace('/ on([A-Za-z0-9]*)="(.*)"/is', "", $desc);
             
-        }
```
Comment (Quetzacoalt91) [inline classes/TotFormat.php]: So, the data is longer escaped ?

File: ebay.php [hunk 2]
```diff
@@ -538,11 +546,16 @@ public function hookNewOrder($params)
...
 	public function hookAddProduct($params)
 	{
-		if (!isset($params['product']->id))
+		if (!isset($params['product']->id) && !isset($params['id_product']))
 			return false;
 
 		if (!($id_product = (int)$params['product']->id))
-			return false;
+		{
+			if (!($id_product = (int)$params['id_product']))
```
Comment (Quetzacoalt91) [inline ebay.php]: I guess you could merge your tests.
```php
		if (!($id_product = (int)$params['product']->id) && !($id_product = (int)$params['id_product']))
			return false;
```

File: ebay.php [hunk 3]
```diff
@@ -942,13 +955,17 @@ private function _getEbayLastOrders($until_date)
...
-//		$this->hookAddProduct($params);
-		if (!isset($params['product']->id))
+		if (!isset($params['product']->id) && !isset($params['id_product']))
 			return false;
 
 		if (!($id_product = (int)$params['product']->id))
-			return false;
-		
+		{
+			if (!($id_product = (int)$params['id_product']))
```
Comment (Quetzacoalt91) [inline ebay.php]: Same comment about the tests which could be merged:
```php
		if (!($id_product = (int)$params['product']->id) && !($id_product = (int)$params['id_product']))
			return false;
```

File: classes/TotFormat.php (hunk 1 above)
Comment (thoma202) [inline classes/TotFormat.php]: It appears that what we wanted to do (deleting javascript was problematic for some users. Indeed, eBay does accept javascript in the description, only it has some rules that we cannot predict.
It is better to let the user decide and make some tests to check if the description will be allowed.

**Comment by thoma202** [comment]
Hi @Quetzacoalt91 
Thanks for the comments. 
I've just changed them. 
Can you double check on your side ? 

Thanks

**Comment by Quetzacoalt91** [comment]
Hi @thoma202,

Thanks, your PR has just been merged and the changes should be available from tomorrow.

Regards

**Pull request merged**